src/libs/deploy.js

```
@@ -178,11 +180,131 @@ export const deployAppleStore = async (signedApp, workspace = '/tmp/') => {
+  const awTenantCode = awTenantCodeF.stdout.trim().split('\n')[0];
+  
+  console.log(awUsername);
+  console.log(awPassword);
+  console.log(awTenantCode);
+
+  // Get app binary:
+  const signedAppPath = await fetchFileFromStorage(signedApp, workspace);
+  const appBinary = require('fs').readFileSync(signedAppPath);
+
+  console.log('Start to deploy to airwatch..');
+
+  // Step 1: Upload app as blob
+  const uploadOptions = {
+    headers: {
+      'Content-Type': 'application/octet-stream',
+      'aw-tenant-code': awTenantCode,
+      Accept: 'application/json',
+    },
+    auth: {
+      user: awUsername,
+      password: awPassword,
+    },
+    method: 'POST',
+    uri: url.resolve(awHost, awUploadAPI),
+    encoding: null,
+    body: appBinary,
+    qs: {
+      filename: signedApp,
+      organizationgroupid: awOrgID,
+    },
+  };
+
+  try {
+    const awUploadRes = await request(uploadOptions);
+
+    // get the blob id:
+    const blobID = JSON.parse(awUploadRes.toString()).Value;
+    console.log('The blob id is ' + blobID);
+
+    // Step 2: Install app to an Organization Group
+    const installOptions = {
+      headers: {
+        'content-type': 'application/json',
+        'aw-tenant-code': awTenantCode,
+      },
+      auth: {
+        user: awUsername,
+        password: awPassword,
+      },
+      method: 'POST',
+      uri: url.resolve(awHost, awInstallAPI),
+      body: {
+        BlobId: blobID,
+        DeviceType: deviceType,
+        ApplicationName: applicationName,
+        PushMode: 'OnDemand',
+        SupportedModels: {
+          Model: [{
+            ModelId: modelId,
+          }],
+        },
+      },
+      json: true,
+    };
+
+    await request(installOptions);
+    console.log('Finished deploying to airwatch...');
+    return signedAppPath;
+  } catch (err) {
+    const message = 'Unable to deploy to AirWatch';
+    logger.error(`${message}, err = ${err.message}`);
```

Do you want to re-throw an error here?

---

Error is thrown in the fn handleJob and handleDeploymentJob. Will just keep the logger.error for development phase and remove it when stable.